bundles/org.openhab.binding.goveelan/src/main/java/org/openhab/binding/goveelan/internal/GoveeLanHandler.java

```diff
@@ -0,0 +1,390 @@
+    private static final int SENDTODEVICE_PORT = 4003;
+    private static final int RECEIVEFROMDEVICE_PORT = 4002;
+
+    // Semaphores to suppress further processing if already running
+    private static boolean refreshJobRunning = false;
+    private static boolean refreshRunning = false;
+
+    @Nullable
+    private static ScheduledFuture<?> refreshStatusJob; // device response receiver job
+    @Nullable
+    private ScheduledFuture<?> triggerStatusJob; // send device status update job
+
+    /*
+     * Common Receiver job for the status answers of the devices
+     */
+    public static boolean isRefreshJobRunning() {
+        return refreshJobRunning && THING_HANDLERS.isEmpty();
+    }
+
+    public static final Gson GSON = new Gson();
+    private static final Runnable REFRESH_STATUS_RECEIVER = () -> {
+        final Logger LOGGER = LoggerFactory.getLogger(GoveeLanHandler.class);
+        /*
+         * This thread receives an answer from any device.
+         * Therefore it needs to apply it to the right thing
+         * 
+         * Discovery uses the same response code, so we must not refresh the status during discovery
+         */
+        if (GoveeLanDiscoveryService.isDiscoveryActive()) {
+            LOGGER.debug("Not running refresh as Scan is currently active");
+        }
+
+        refreshJobRunning = true;
+        LOGGER.trace("REFRESH: running refresh cycle for {} devices", THING_HANDLERS.size());
+
+        if (THING_HANDLERS.isEmpty()) {
+            return;
+        }
+
+        GoveeLanHandler thingHandler = null;
+        try (MulticastSocket socket = new MulticastSocket(RECEIVEFROMDEVICE_PORT)) {
+            byte[] buffer = new byte[10240];
+            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+            socket.setReuseAddress(true);
+            LOGGER.debug("waiting for Status");
+            socket.receive(packet);
+
+            String response = new String(packet.getData()).trim();
+            String deviceIPAddress = packet.getAddress().toString().replace("/", "");
+            LOGGER.trace("received = {} from {}", response, deviceIPAddress);
+
+            thingHandler = THING_HANDLERS.get(deviceIPAddress);
+            if (thingHandler == null) {
+                LOGGER.warn("thing Handler for {} couldn't be found.", deviceIPAddress);
+                return;
+            }
+
+            LOGGER.debug("updating status for thing {} ", thingHandler.getThing().getLabel());
+            LOGGER.trace("Response from {} = {}", deviceIPAddress, response);
+
+            if (response != null && !response.isEmpty()) {
+                try {
+                    StatusMessage statusMessage = GSON.fromJson(response, StatusMessage.class);
+                    thingHandler.updateDeviceState(statusMessage);
+                } catch (JsonSyntaxException jse) {
+                    thingHandler.updateStatus(ThingStatus.OFFLINE);
```

Comment:
This can be considered as COMMUNICATION_ERROR since the response is unexpected.
```suggestion
                    thingHandler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, jse.getMessage());
```